File: vitrola.server/src/models/songRequest.ts
import { Column, Entity, PrimaryGeneratedColumn } from 'typeorm';

@Entity()
export class SongRequest {
  @PrimaryGeneratedColumn('uuid')
  id?: string;
  @Column()
  songRequestId: string;
  @Column({ nullable: true })
  songId?: string | null;
  @Column()
  songRequestDate: Date;
  @Column()
  isForAddingNewSong: boolean;
  @Column()
  isForPlayingSong: boolean;
  @Column({ nullable: true })
  songName?: string | null;
  @Column({ nullable: true })
  songArtist?: string | null;
}

File: Vitrola.Mobile/Vitrola.Mobile/Models/SongRequest.swift
//
//  SongRequest.swift
//  Vitrola.Mobile
//
//  Created by Natanael Ureña Fernández  on 12/3/23.
//

import Foundation

struct SongRequest: Codable {
    var songRequestId: String?
    var songId: String?
    var songName: String?
    var songArtist: String?
    var isForAddingNewSong: Bool
    var isForPlayingSong: Bool
    var songRequestDate: Date
}
